src/main/java/org/researchspace/sail/rest/RESTSailConnection.java

```diff
@@ -0,0 +1,267 @@
+
+    public RESTSailConnection(AbstractServiceWrappingSail<RESTSailConfig> sailBase) {
+        super(sailBase);
+    }
+
+    @Override
+    protected Collection<BindingSet> convertStream2BindingSets(InputStream inputStream,
+            RESTParametersHolder parametersHolder) throws SailException {
+
+        List<BindingSet> results = Lists.newArrayList();
+
+        try {
+            String stringResponse = IOUtils.toString(inputStream, StandardCharsets.UTF_8.name());
+            Model model = getSail().getServiceDescriptor().getModel();
+
+            // TODO: check the string format and call the right type. E.g., json or XML
+            String type = "json";
+            switch (type) {
+            case "json":
+                // Get a list of output variable names and their json paths
+                Map<String, String> jsonPaths = getJsonPaths(parametersHolder.getOutputVariables(), model);
+
+                // Get the root path
+                Optional<Parameter> param = getSail().getSubjectParameter();
+                String rootPath = param.isPresent() ? getJsonPath(param.get(), model) : "$";
+
+                results = executeJson(stringResponse, jsonPaths, rootPath);
+                break;
+
+            default:
+                break;
+            }
+
+            return results;
+        } catch (Exception e) {
+            throw new SailException(e);
+        }
+    }
+
+    /**
+     * 
+     * @param res
+     * @param parametersHolder
+     * @param model
+     */
+    private List<BindingSet> executeJson(String res, Map<String, String> jsonPaths, String rootPath) {
+
+        // Parse the response
+        ReadContext context = JsonPath.parse(res);
+
+        // Get the root object
+        Object root = context.read(rootPath);
+
+        List<BindingSet> results = Lists.newArrayList();
+
+        // Manipulate the results in different ways based on its structure
+        if (root instanceof JSONArray)
+            results = iterateJsonArray((JSONArray) root, jsonPaths);
+
+        if (root instanceof Map)
+            results = iterateJsonMap((Map) root, jsonPaths);
+
+        return results;
+    }
+
+    /**
+     * 
+     * @param array
+     * @param outputVariables
+     * @param model
+     * @return
+     */
+    private List<BindingSet> iterateJsonArray(JSONArray array, Map<String, String> jsonPaths) {
+
+        List<BindingSet> bindingSets = Lists.newArrayList();
+
+        // Iterate over each object in JSONArray and evaluate its
+        for (Object object : array) {
+            MapBindingSet mapBindingSet = new MapBindingSet();
+
+            // Evaluate each jsonPath singularly and add the result to the binding
+            for (Map.Entry<String, String> path : jsonPaths.entrySet()) {
+                mapBindingSet.addBinding(path.getKey(),
+                        VF.createLiteral(JsonPath.read(object, path.getValue()).toString(), XSD.STRING));
```

Comment:
we shouldn't always use XSD.STRING, maybe as a default is OK, but if defined we should take the value from `spl:valueType`, e.g as in:
```
    spin:column [
        a spin:Column ;
        rdfs:comment "Display Name" ;
        spl:predicate :_display_name ;
        spl:valueType xsd:string ;
        ephedra:jsonPath "$.display_name"
        ] ;
```

Reply:
See https://github.com/researchspace/researchspace/commit/89b5639bc36a559d8c701105a087ee8e321ead70